x-pack/plugins/maps/public/classes/styles/vector/vector_style.tsx

```diff
@@ -239,11 +241,173 @@ export class VectorStyle implements IVectorStyle {
+          nextField,
+          invalidStyleName
+        );
+        if (!isFieldDataTypeCompatible) {
+          return;
+        }
+        const newFieldDescriptor = rectifyFieldDescriptor(nextField as IESAggField, {
+          origin: previousField.getOrigin(),
+          name: previousField.getName(),
+        });
+        hasChanges = true;
+        (originalProperties[invalidStyleName]!
+          .options! as DynamicStylePropertyOptions).field = newFieldDescriptor;
+      }
+    });
+
+    return {
+      hasChanges,
+      nextStyleDescriptor: VectorStyle.createDescriptor(originalProperties, this.isTimeAware()),
+    };
+  }
+
+  async _deleteFieldsFromDescriptorAndUpdateStyling(
+    nextFields: IField[],
+    styleFieldsHelper: StyleFieldsHelper,
+    originalProperties: VectorStylePropertiesDescriptor,
+    mapColors: string[],
+    hasChanges: boolean
+  ) {
+    const updatedProperties = {} as VectorStylePropertiesDescriptor;
+
+    const dynamicProperties = (Object.keys(originalProperties) as VECTOR_STYLES[]).filter((key) => {
+      const dynamicOptions = getDynamicOptions(originalProperties, key);
+      return dynamicOptions && dynamicOptions.field && dynamicOptions.field.name;
+    });
+
+    dynamicProperties.forEach((key: VECTOR_STYLES) => {
+      // Convert dynamic styling to static stying when there are no style fields
+      const styleFields = styleFieldsHelper.getFieldsForStyle(key);
+      if (styleFields.length === 0) {
+        const staticProperties = getDefaultStaticProperties(mapColors);
+        updatedProperties[key] = staticProperties[key] as any;
+        return;
+      }
+
+      const dynamicProperty = originalProperties[key];
+      if (!dynamicProperty || !dynamicProperty.options) {
+        return;
+      }
+      const fieldName = (dynamicProperty.options as DynamicStylePropertyOptions).field!.name;
+      if (!fieldName) {
+        return;
+      }
+
+      const matchingOrdinalField = nextFields.find((ordinalField) => {
+        return fieldName === ordinalField.getName();
+      });
+
+      if (matchingOrdinalField) {
+        return;
+      }
+
+      updatedProperties[key] = {
+        type: DynamicStyleProperty.type,
+        options: {
+          ...originalProperties[key]!.options,
+        },
+      } as any;
+
+      if ('field' in updatedProperties[key].options) {
+        delete (updatedProperties[key].options as DynamicStylePropertyOptions).field;
+      }
+    });
+
+    if (Object.keys(updatedProperties).length !== 0) {
+      return {
+        hasChanges: true,
+        nextStyleDescriptor: VectorStyle.createDescriptor(
+          {
+            ...originalProperties,
+            ...updatedProperties,
+          },
+          this.isTimeAware()
+        ),
+      };
+    } else {
+      return {
+        hasChanges,
+        nextStyleDescriptor: VectorStyle.createDescriptor(originalProperties, this.isTimeAware()),
```

Comment:
Just return `nextStyleDescriptor: { ...this._descriptor }` since there are no changes to the descriptor.